File: 2/rust/src/main.rs
fn main() {
    let mut sum = 2;
    let mut numbers = vec![1, 2];
    let mut num = 0;
    while num < 4_000_000 {
        let num1 = numbers.get(numbers.len() - 2);
        let num2 = numbers.get(numbers.len() - 1);
        if num1.is_none() || num2.is_none() {
            eprintln!("Error: numbers.get(numbers.len() - 2) or numbers.get(numbers.len() - 1) is None");
            break;
        } else {
            num = num1.unwrap() + num2.unwrap();
            numbers.push(num);

            if num & 2 == 0 {
                sum += num;
            }
            println!("num: {}", num);
        }
    }
    println!("sum: {}", sum);
}

File: 2/java/src/main/java/Main.java
public class Main {
    public static void main(String[] args) {
        int num1 = 0;
        int num2 = 1;
        long sum = 0;

        while (num2 < 4_000_000) {
            int fibonacci = num1 + num2;
            num1 = num2;
            num2 = fibonacci;

            if (fibonacci % 2 == 0) {
                sum += fibonacci;
            }
        }
        System.out.println(sum);
    }
}
